### PR TITLE
[JENKINS-28022] Make descriptions copy-able without zero-width spaces

### PR DESCRIPTION
**Pull request by gdubicki**
as they cause obscure issues almost everywhere where they are used.
At the same time simplify the code substantially.
    
*Update*: thx to @hakubo for providing the target solution using `word-break: break-word;`.

## Proposed changelog entries

* [JENKINS-28022] - Make descriptions in the Build History widget copy-able without zero-width spaces

### CODE REVIEW

**Comment by daniel-beck** [comment]
Thanks for your work on this problem. Unfortunately, this code exists for a reason, and this PR doesn't even attempt to address that use case.

## Before (2.249.2)

Screenshot

## After (PR build)

Screenshot

**Review by daniel-beck — CHANGES_REQUESTED** [review]
As above.

**Comment by gdubicki** [comment]
Hey, sorry for that obvious minimal-effort first approach. 😅 My real idea for this is to replace the zws with a <wbr/> tag. I will try and test that.

**Comment by daniel-beck** [comment]
@gdubicki Yes, sadly it's not quite that easy. It might still be easy (some CSS perhaps?) but not _this_ easy 😁

**Comment by martinda** [comment]
Work on this was presented and discussed in https://github.com/jenkinsci/jenkins/pull/3377, but was later abandoned. CSS was recommended there too.

**Comment by fredericrous** [comment]
Looking forward for a fix, keep up the good work! In my opinion having the ui look broken when there is long strings is bad, sure, but it is a lower bug than having an invisible character inserted in the version number

**Comment by gdubicki** [comment]
We also remove that character in some places already, @fredericrous , but we are being hit by this over and over in new places every now and then. So we would like to finally fix the root cause of the issue.

**Comment by gdubicki** [comment]
Thanks to @hakubo I fixed the code to use `word-break: break-word;` CSS property.

That property is supported by 96-98% of all the browers, according to https://caniuse.com/?search=word-break .

Preview: 
Screenshot 2020-10-22 at 17 19 11

Is this ok everyone?

**Comment by gdubicki** [comment]
@daniel-beck , @oleg-nenashev : a friendly reminder that this PR awaits your review :)

**Comment by gdubicki** [comment]
...and another gentle ping for a review. :)

**Comment by primoz-p** [comment]
> Looking forward for a fix, keep up the good work! In my opinion having the ui look broken when there is long strings is bad, sure, but it is a lower bug than having an invisible character inserted in the version number.
> 
> Anyway, the approach we are taking in my service is to accept that jenkins adds an extra invisible character in the UI and "fix" this by removing this extra character where it is used. As it happens, it's another jenkins job that takes the build version as parameter. I haven't tested this code because I'm not in charge of the resolution but this should do the trick for anyone reading this
> 
> ```
> String versionToUse = params.version.replaceAll("[\\p{Cf}]", "");
> ```

Same problem here.
We spent too much time today searching for a cause of failed build. Later we found out that somebody just copied version from build description and that there was this invisible character at the beginning of parameter.

Hoping you'll find a solution and I'm sure that "technology" was updated from 2014/15 where this zero-space-character was used for work wrapping...

**Review by daniel-beck — APPROVED** [review]
Tested interactively using the core PR tester, and it worked for my test cases.

Screenshot

This looks like a reasonable improvement.

**Comment by gdubicki** [comment]
Great, thanks @daniel-beck !

Can you take a look too please, @oleg-nenashev ?

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
Not sure whether it will work fine for any localization, but a default CSS property looks much better than the manual word break implementation. Thanks for the patch!

**Comment by jsoref** [comment]
Fwiw, in theory, I think one should try to use [break-all](https://caniuse.com/?search=break-all) ([ref](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#values)) value and fall back to `break-word`

**Pull request merged**
